packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -150,6 +150,80 @@ const useTransition = (
   return { placeholder: {}, image: {} };
 };
 
+const findBestSourceForSize = (
+  sources: ImageSource[] | undefined,
+  size: DOMRect | null
+): ImageSource | null => {
+  return (
+    sources
+      // look for the smallest image that's still larger then a container
+      ?.map((source) => {
+        if (!size) return { source, penalty: 0, covers: false };
+        const { width, height } =
+          typeof source === 'object' ? source : { width: null, height: null };
+        if (width == null || height == null) return { source, penalty: 0, covers: false };
+        if (width < size.width || height < size.height)
+          return {
+            source,
+            penalty: Math.max(size.width - width, size.height - height),
+            covers: false,
+          };
+        return { source, penalty: (width - size.width) * (height - size.height), covers: true };
+      })
+      .sort((a, b) => a.penalty - b.penalty)
+      .sort((a, b) => Number(b.covers) - Number(a.covers))[0]?.source ?? null
```

Comment:
`.sort` in JS is in-place, so `findBestSourceForSize` has a side effect but its name doesn't suggest that 😅 I think it's better to copy the array and keep the input unsorted.